src/run.ts

```diff
@@ -50,20 +51,37 @@ export async function run() {
          runnerTempDirectory,
          `kubeconfig_${Date.now()}`
       )
+      let cmd = []
       core.debug(`Writing kubeconfig to ${kubeconfigPath}`)
-      const cmd = [
-         'aks',
-         'get-credentials',
-         '--resource-group',
-         resourceGroupName,
-         '--name',
-         clusterName,
-         '-f',
-         kubeconfigPath
-      ]
-      if (subscription) cmd.push('--subscription', subscription)
-      if (admin) cmd.push('--admin')
-      if (publicFqdn) cmd.push('--public-fqdn')
+      if (fleetName){
+         cmd = [
+            'fleet',
```

Comment:
Can we do something like this
```typescript
cmd = [
  (fleetName)? 'fleet' : 'aks',
  ...
```

And then lines 81 and 82 can be:
```typescript
if (!fleet) {
  if (admin) cmd.push ('--admin')
  if (publicFqdn) cmd.push ('--public-fqdn')
}
```

I'm not an expert in TS but might be worth a try.

Reply:
good catch, will do that strategy with fleet and cluster name as well!